### DISCUSSION TITLE
Combine "is" and "is not" filter for pages

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#2795] [Anything Else]
"Any chance that it'll be possible in the future to combine "is" and "is not" filters? That'd be helpful to filter out specific pages when selecting a group..."

(idea from https://twitter.com/danisager/status/1640394334359941120)

**Comment by dsager** [comment]
The idea was proposed by me on Twitter, following up with a concrete example:

We have several URLs under `/news/*` that we want to look at while excluding others. All our news articles have the format `/news/foo-bar-123`, but there are also other URLs like `/news/search`. So if we need to look at traffic for all our news articles, I'd like to filter for `page is '/news/*' && page is not '/news/search`.

Another approach would be to support regular expressions, as proposed in https://github.com/plausible/analytics/discussions/1890

**Reply by ukutaht** [reply]
gotcha. You're clearly a dev so I'll share the details. The filters are represented as a hashmap data type, both in the frontend and backend. It would require a big refactor to be able to support more than one filter expression per property like `page`.

In the short term, we are planning to start testing the new [custom properties](https://github.com/plausible/analytics/discussions/320) feature. This could work well for your use-case:

1. On the pages you want to group together (assuming it's a template for `/news/:id` or similar), you could add a custom property to these pageviews, something like `pageGroup="news"`
2. Then in the Plausible dashboard you'll be able to add a filter for `props.pageGroup is 'news'`. The dashboard will be filtered for that group of pages

I understand this is not optimal, ideally you wouldn't have to deploy or make changes to your website in order to achieve this. But hey, until we get to those more polished features this could be a decent workaround for you.

**Reply by dsager** [reply]
Thanks for the additional context and the proposed work around. I'm looking forward to custom properties for a lot of other use cases already ;)

Quick thought regarding the filtering logic, would it be possible to negate per value in the URL parameter?
Currently `?page=!/foo|/bar` is interpreted as `is not /foo && is not /bar`. What if you changed the logic so that:
- `?page=!/foo|/bar` is interpreted as `is not /foo && is /bar`
- `?page=!/foo|!/bar` is interpreted as `is not /foo && is not /bar`

I guess there are other complications I'm not seeing, just a thought...

Regarding regex filtering (that would be my favorite solution) - I understand that this is a limitation by the data store that you're using? So unlikely to come at some point?

**Reply by ukutaht** [reply]
> What if you changed the logic so that:

Yeah, that could work while keeping the changes smaller.

> Regarding regex filtering (that would be my favorite solution) - I understand that this is a limitation by the data store that you're using? So unlikely to come at some point?

Actually no, we can use regex matches in the database we use.

Really the main blocker is the small amount of people requesting for this at the moment. Comparatively we have much bigger fish to fry like comparisons, custom props, etc. which we are actively working on. I'm sure we will expand the filtering capabilities in the future but I think at the moment we'll need to focus on other priorities.

**Reply by dsager** [reply]
Thanks for the info, that makes total sense. Keep up the good work!
